mesonbuild/backend/ninjabackend.py

```diff
@@ -2002,6 +2002,10 @@ def get_build_dir_include_args(self, target, compiler):
             curdir = '.'
         return compiler.get_include_args(curdir, False)
 
+    @lru_cache(maxsize=None)
+    def get_normpath_target(self, source):
+        return os.path.normpath(source)
```

Comment:
```suggestion
def get_normpath_target(self, source: str) -> str:
        return os.path.normpath(source)
```

Please use type annotations for the new functions if possible.